publisher/options.go

```diff
@@ -0,0 +1,57 @@
+package publisher
+
+import (
+	"strconv"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+// Option allows to configure RabbitMQ Publisher.
+type Option func(p *Publisher)
+
+// WithConstHeaders sets constant publishing headers, that will be added to all publishings.
+func WithConstHeaders(headers amqp.Table) Option {
```

Comment:
What about adding a similar method that would allow to set middlewares? For example when generating a message ID as UUID ;-)

Reply:
this can be done with per-publish middleware, see https://github.com/heureka/gorabbit/blob/feat/publisher/examples/publisher/main.go#L33